### ISSUE TITLE
Fatal runtime error (panic) when attempting to edit invalid symlinks.

### ISSUE THREAD

**Issue by gburtini** [justwatchcom/gopass#369]
I accidentally set up a broken symlink (to an invalid path) and instead of catching this graceully, gopass v. 1.3.2 triggers a panic upon edit.

```
$ gopass edit [path-to-broken-symlink]
panic: runtime error: slice bounds out of range

goroutine 1 [running]:
github.com/justwatchcom/gopass/fsutil.CleanPath(0xc420018300, 0x3e, 0x3a, 0x1461bf2)
	/private/tmp/gopass-20170919-13288-771sxw/gopass-1.3.2/src/github.com/justwatchcom/gopass/fsutil/fsutil.go:22 +0x21e
github.com/justwatchcom/gopass/store/sub.(*Store).passfile(0xc42005cc60, 0x7ffeefbff9c4, 0x1d, 0x0, 0x0)
	/private/tmp/gopass-20170919-13288-771sxw/gopass-1.3.2/src/github.com/justwatchcom/gopass/store/sub/store.go:506 +0xde
github.com/justwatchcom/gopass/store/sub.(*Store).Exists(0xc42005cc60, 0x7ffeefbff9c4, 0x1d, 0x0)
	/private/tmp/gopass-20170919-13288-771sxw/gopass-1.3.2/src/github.com/justwatchcom/gopass/store/sub/store.go:250 +0x43
github.com/justwatchcom/gopass/store/root.(*Store).Exists(0xc420118100, 0x7ffeefbff9c4, 0x1d, 0x116f5ce)
	/private/tmp/gopass-20170919-13288-771sxw/gopass-1.3.2/src/github.com/justwatchcom/gopass/store/root/store.go:223 +0x93
github.com/justwatchcom/gopass/action.(*Action).Edit(0xc420118080, 0xc420092b40, 0xc4201149d8, 0xc4201149d8)
	/private/tmp/gopass-20170919-13288-771sxw/gopass-1.3.2/src/github.com/justwatchcom/gopass/action/edit.go:27 +0x91
github.com/justwatchcom/gopass/action.(*Action).Edit-fm(0xc420092b40, 0x0, 0x0)
	/private/tmp/gopass-20170919-13288-771sxw/gopass-1.3.2/src/github.com/justwatchcom/gopass/main.go:255 +0x34
github.com/justwatchcom/gopass/vendor/github.com/urfave/cli.HandleAction(0x13e08a0, 0xc42004f520, 0xc420092b40, 0xc42005cd00, 0x0)
	/private/tmp/gopass-20170919-13288-771sxw/gopass-1.3.2/src/github.com/justwatchcom/gopass/vendor/github.com/urfave/cli/app.go:485 +0xd2
github.com/justwatchcom/gopass/vendor/github.com/urfave/cli.Command.Run(0x1461d4e, 0x4, 0x0, 0x0, 0xc42004f840, 0x1, 0x1, 0x1476b9b, 0x3d, 0x0, ...)
	/private/tmp/gopass-20170919-13288-771sxw/gopass-1.3.2/src/github.com/justwatchcom/gopass/vendor/github.com/urfave/cli/command.go:193 +0xa99
github.com/justwatchcom/gopass/vendor/github.com/urfave/cli.(*App).Run(0xc42007d040, 0xc42000e0c0, 0x3, 0x3, 0x0, 0x0)
	/private/tmp/gopass-20170919-13288-771sxw/gopass-1.3.2/src/github.com/justwatchcom/gopass/vendor/github.com/urfave/cli/app.go:250 +0x758
main.main()
	/private/tmp/gopass-20170919-13288-771sxw/gopass-1.3.2/src/github.com/justwatchcom/gopass/main.go:603 +0x4dfa
```

Perhaps we deem this acceptable as my symlink was me manually editing the password-store directory (and screwing it up), but I generally do not expect cryptographic software to crash unexpectedly in most cases. 

For what it is worth, `pass` behaves as arguably expected (?) and returns an empty file.

**Comment by martinhoefling** [comment]
This is probably due to https://github.com/justwatchcom/gopass/blob/master/utils/fsutil/fsutil.go#L20 . The function CleanPath does not return any error and the error is not handled but path is assigned. One option would be to assign path only if resolving the symlink returned no error -> see #370.

**Comment by dominikschulz** [comment]
We use `panic` deliberately in a few places if we can't figure out a safe way to continue, but is none of them.

**Comment by dominikschulz** [comment]
Should be fixed by #370
